Tella/Scenes/Resources/Models/Resource.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  Resource.swift
+//  Tella
+//
+//  Created by gus valbuena on 2/1/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+
+struct Resource: Codable, Identifiable {
+    let id: String
+    let title: String
+    let fileName: String
+    let size: String
+    let createdAt: String
+
+    enum CodingKeys: String, CodingKey {
+        case id, title, fileName, size, createdAt
+    }
+}
+
+struct DownloadedResource: Codable, Identifiable {
+    let id: String
+    let externalId: String
+    let title: String
+    let fileName: String
+    let size: String
+    let serverId: Int?
```

Comment:
I think it's better to add the `server` instead of  `serverId`
see `func getReports(reportStatus:[ReportStatus]) -> [Report]`  to see the `joinCondition`

Reply:
so, after adding server to the resource class I will have to modify the Server class to be decodable right? 
or should I decoded the server properties manually on the resource class? bringing this up because I already implement the decode function to parse the array when fetching the resources from the db so I'm curious to know what should be the best approach here

Reply:
It's better to use the decode function to parse the ressources, so you can try to modify the server to Decodable
In the `DownloadedResource` class  I think ` init(from decoder: Decoder) throws` is unuseful

Reply:
so I change it but i couldn't fin a way to automatically decode the server after getting the query response. When joining both tables I receive a flat object and the DownloadResource class is expecting a nested server object. So I had to decode manually in the DownlodResource model. 

We could try to parse the server inside the query but I think we will over complex the database function and we will be missing the point of using decode to make the call simpler. 

Anyway let me know what do you think and if you know a better way to decode it!

Reply:
I suggest :

```
    func getDownloadedResources() -> [DownloadedResource] {
        
        do {
            let joinCondition = [JoinCondition(tableName: D.tServer,
                                               firstItem: JoinItem(tableName: D.tResource, columnName: D.cServerId),
                                               secondItem: JoinItem(tableName: D.tServer, columnName: D.cServerId))]
            let responseDict = try statementBuilder.selectQuery(tableName: D.tResource, joinCondition: joinCondition)

            let decodedResources = try responseDict.compactMap({ dict in
                let ressource = try dict.decode(DownloadedResource.self)
                let server = try dict.decode(Server.self)
                ressource.server = server
                return ressource
            })
            
            return decodedResources
        } catch {
            return []
        }
    }


```

and 

```
class DownloadedResource: Codable, Identifiable {
    var id: String
    var externalId: String
    var title: String
    var fileName: String
    var size: String
    var createdAt: String
    var server: Server?
    
    enum CodingKeys: String, CodingKey {
        case id = "c_id"
        case externalId = "c_external_id"
        case title = "c_title"
        case fileName = "c_filename"
        case size = "c_size"
        case createdAt = "c_created_date"
    }
}

```


because the dictionnary from the db is a dictionnary with only level so the server won't be recognized